File: FirmaTransportSocketsJava/Networking/src/main/java/dto/DTOUtils.java
package dto;

import domain.Cursa;
import domain.Oficiu;
import domain.Rezervare;

import java.time.LocalDateTime;

public class DTOUtils {
    public static Oficiu getFromDTO(OficiuDTO ofdto){
        Integer id =ofdto.getId();
        String nume= ofdto.getNume();;
        String prenume = ofdto.getParola();
        Oficiu oficiu=new Oficiu(nume,prenume);
        oficiu.setId(id);
        return oficiu;
    }

    public static OficiuDTO getDTO(Oficiu of){
        Integer id =of.getId();
        String nume= of.getNume();;
        String prenume = of.getParola();
        return new OficiuDTO(id,nume,prenume);
    }

    public static Oficiu[] getFromDTO(OficiuDTO[] ofdtos){
        Oficiu[] oficii=new Oficiu[ofdtos.length];
        for(int i=0; i<ofdtos.length;i++){
            oficii[i]=getFromDTO(ofdtos[i]);
        }
        return oficii;
    }

    public static OficiuDTO[] getDTO(Oficiu[] ofs){
        OficiuDTO[] oficii=new OficiuDTO[ofs.length];
        for(int i=0; i<ofs.length;i++){
            oficii[i]=getDTO(ofs[i]);
        }
        return oficii;
    }

    public static Cursa getFromDTO(CursaDTO csdto){
        Integer id=csdto.getId();
        String dest= csdto.getDestinatie();
        LocalDateTime plec =csdto.getPlecare();
        Integer nrl=csdto.getLocuriDisponibile();
        Cursa cursa=new Cursa(dest,plec,nrl);
        cursa.setId(id);
        return cursa;
    }

    public static CursaDTO getDTO(Cursa cs){
        Integer id=cs.getId();
        String dest= cs.getDestinatie();
        LocalDateTime plec =cs.getPlecare();
        Integer nrl=cs.getLocuriDisponibile();

        return new CursaDTO(id,dest,plec,nrl);
    }

    public static Cursa[] getFromDTO(CursaDTO[] csdtos){
        Cursa[] curse=new Cursa[csdtos.length];
        for(int i=0;i<csdtos.length;i++){
            curse[i]=getFromDTO(csdtos[i]);
        }
        return curse;
    }

    public static CursaDTO[] getDTO(Cursa[] css){
        CursaDTO[] curse=new CursaDTO[css.length];
        for(int i=0;i<css.length;i++){
            curse[i]=getDTO(css[i]);
        }
        return curse;
    }

    public static Rezervare getFromDTO(RezervareDto rezdto){
        Integer id=rezdto.getId();
        String numeClient= rezdto.getNumeClient();
        Integer nrL = rezdto.getNrLocuri();
        Integer curs=rezdto.getIdCursa();
        Rezervare rezervare=new Rezervare(numeClient,nrL,curs);
        rezervare.setId(id);
        return rezervare;
    }

    public static domain.RezervareDTO getDTOFromDTO(RezervareDto rezdto){
        String numeClient= rezdto.getNumeClient();
        Integer nrL = rezdto.getNrLocuri();
        return new domain.RezervareDTO(numeClient,nrL);
    }

    public static RezervareDto getDTO(Rezervare rez){
        Integer id=rez.getId();
        String numeClient= rez.getNumeClient();
        Integer nrL = rez.getNrLocuri();
        Integer curs=rez.getIdCursa();
        return new RezervareDto(id,numeClient,nrL,curs);
    }


    public static RezervareDto getDTOdom(domain.RezervareDTO rez){
        //Integer id=rez.getId();
        String numeClient= rez.getNumeClient();
        Integer nrL = rez.getNrLoc();
        //Integer curs=rez.getIdCursa();
        return new RezervareDto(numeClient,nrL);
    }


    public static Rezervare[] getFromDTO(RezervareDto[] rezdtos){
        Rezervare[] rezervari=new Rezervare[rezdtos.length];
        for(int i=0;i< rezdtos.length;i++){
            rezervari[i]=getFromDTO(rezdtos[i]);
        }
        return rezervari;

    }

    public static RezervareDto[] getDTO(Rezervare[] rezs){
        RezervareDto[] rezervari=new RezervareDto[rezs.length];
        for(int i=0;i< rezs.length;i++){
            rezervari[i]=getDTO(rezs[i]);
        }
        return rezervari;

    }

    public static domain.RezervareDTO[] getDTOFromDTO(RezervareDto[] rezdtos){
        domain.RezervareDTO[] rezervari=new domain.RezervareDTO[rezdtos.length];
        for(int i=0;i< rezdtos.length;i++){
           rezervari[i]=getDTOFromDTO(rezdtos[i]);
        }
        return rezervari;

    }

    public static RezervareDto[] getDTOdom(domain.RezervareDTO[] rezs){
        RezervareDto[] rezervari=new RezervareDto[rezs.length];
        for(int i=0;i< rezs.length;i++){
            rezervari[i]=getDTOdom(rezs[i]);
        }
        return rezervari;

    }




}

File: FirmaTransportSocketsC#/Networking/DTOUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Model;

namespace Networking.dto
{
    public class DTOUtils
    {
        public static Oficiu getFromDTO(OficiuDto oficiuDto)
        {
            int id = oficiuDto.id;
            string nume = oficiuDto.nume;
            string parola = oficiuDto.parola;

            return new Oficiu(id,nume,parola);
        }

        public static OficiuDto getDTO(Oficiu oficiu)
        {
            int id = oficiu.id;
            string nume = oficiu.nume;
            string parola = oficiu.parola;

            return new OficiuDto(id, nume, parola);
        }


        public static Oficiu[] getFromDTO(OficiuDto[] oficiuDto)
        {
            Oficiu[] oficii = new Oficiu[oficiuDto.Length];
            for(int i = 0; i < oficiuDto.Length; i++)
            {
                oficii[i] = getFromDTO(oficiuDto[i]);
            }

            return oficii;
        }


        public static OficiuDto[] getDTO(Oficiu[] oficiu)
        {
            OficiuDto[] oficii = new OficiuDto[oficiu.Length];
            for (int i = 0; i < oficiu.Length; i++)
            {
                oficii[i] = getDTO(oficiu[i]);
            }

            return oficii;
        }

        //--------------


        public static Cursa getFromDTO(CursaDto cursaDto)
        {
            int id = cursaDto.id;
            string destinatie = cursaDto.destinatie;
            DateTime plecare = cursaDto.plecare;
            int locuri = cursaDto.locuriDisponibile;
            return new Cursa(id, destinatie, plecare, locuri);
        }

        public static CursaDto getDTO(Cursa cursa)
        {
            int id = cursa.id;
            string destinatie = cursa.destinatie;
            DateTime plecare = cursa.plecare;
            int locuri = cursa.locuriDisponibile;
            return new CursaDto(id, destinatie, plecare, locuri);
        }

        public static Cursa[] getFromDTO(CursaDto[] cursaDto)
        {
            Cursa[] curse = new Cursa[cursaDto.Length];
            for (int i = 0; i < cursaDto.Length;i++)
            {
                curse[i] = getFromDTO(cursaDto[i]);

            }

            return curse;
        }

        public static CursaDto[] getDTO(Cursa[] cursa)
        {
            CursaDto[] curse = new CursaDto[cursa.Length];
            for (int i = 0; i < cursa.Length; i++)
            {
                curse[i] = getDTO(cursa[i]);

            }

            return curse;
        }

        //---------------------------


        public static Rezervare getFromDTO(RezervareDto rezervareDto)
        {
            int id = rezervareDto.id;
            string nume = rezervareDto.numeClient;
            int idCursa = rezervareDto.idCursa;
            int nrLoc = rezervareDto.nrLoc;
            return new Rezervare(id, nume, nrLoc, idCursa);

        }

        public static RezervareDto getDTO(Rezervare rezervare)
        {
            int id = rezervare.id;
            string nume = rezervare.numeClient;
            int idCursa = rezervare.idCursa;
            int nrLoc = rezervare.nrLocuri;
            return new RezervareDto(id, nume, nrLoc, idCursa);

        }

        public static Rezervare[] getFromDTO(RezervareDto[] rezervareDto)
        {
            Rezervare[] rezervari = new Rezervare[rezervareDto.Length];
            for(int i = 0; i < rezervareDto.Length; i++)
            {
                rezervari[i] = getFromDTO(rezervareDto[i]);
            }
            return rezervari;

        }

        public static RezervareDto[] getDTO(Rezervare[] rezervare)
        {
            RezervareDto[] rezervari = new RezervareDto[rezervare.Length];
            for (int i = 0; i < rezervare.Length; i++)
            {
                rezervari[i] = getDTO(rezervare[i]);
            }
            return rezervari;

        }

        //--------------------


        public static RezervareDTO getDTOFromDTO(RezervareDto rezervareDto)
        {
            string nume = rezervareDto.numeClient;
            int nrLoc = rezervareDto.nrLoc;
            return new RezervareDTO(nume, nrLoc);
        }

        public static RezervareDto getDTOdom(RezervareDTO rezervareDto)
        {
            string nume = rezervareDto.numeClient;
            int nrLoc = rezervareDto.nrLo;
            return new RezervareDto(nume, nrLoc);

        }


        public static RezervareDTO[] getDTOFromDTO(RezervareDto[] rezervareDto)
        {
            RezervareDTO[] rezervares = new RezervareDTO[rezervareDto.Length];
            for(int i = 0; i < rezervareDto.Length; i++)
            {
                rezervares[i] = getDTOFromDTO(rezervareDto[i]);
            }
            return rezervares;
        }


        public static RezervareDto[] getDTOdom(RezervareDTO[] rezervareDto)
        {
            RezervareDto[] rezervares = new RezervareDto[rezervareDto.Length];
            for (int i = 0; i < rezervareDto.Length; i++)
            {
                rezervares[i] = getDTOdom
                    (rezervareDto[i]);
            }
            return rezervares;
        }




    }
}
